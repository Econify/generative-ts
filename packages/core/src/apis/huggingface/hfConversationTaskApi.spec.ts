import {
  HfConversationalTaskApi,
  HfConversationalTaskOptions,
} from "./hfConversationTaskApi";

function render(context: Omit<HfConversationalTaskOptions, "modelId">) {
  try {
    const rendered = HfConversationalTaskApi.requestTemplate.render({
      modelId: "mock-model-id",
      ...context,
    });
    return JSON.parse(rendered);
  } catch (error) {
    throw new Error("Invalid JSON output");
  }
}

describe("HfConversationalTaskOptions:", () => {
  test("prompt", () => {
    const rendered = render({
      prompt: "mock-prompt",
    });

    expect(rendered).toMatchSnapshot();
  });

  test("prompt, past_user_inputs", () => {
    const rendered = render({
      prompt: "mock-prompt",
      past_user_inputs: ["input1", "input2"],
    });

    expect(rendered).toMatchSnapshot();
  });

  test("prompt, generated_responses", () => {
    const rendered = render({
      prompt: "mock-prompt",
      generated_responses: ["response1", "response2"],
    });

    expect(rendered).toMatchSnapshot();
  });

  test("prompt, past_user_inputs, generated_responses", () => {
    const rendered = render({
      prompt: "mock-prompt",
      past_user_inputs: ["input1", "input2"],
      generated_responses: ["response1", "response2"],
    });

    expect(rendered).toMatchSnapshot();
  });

  test("prompt, parameters", () => {
    const rendered = render({
      prompt: "mock-prompt",
      parameters: {
        min_length: 10,
        max_length: 100,
        top_k: 50,
        top_p: 0.9,
        temperature: 0.7,
        repetition_penalty: 1.2,
        max_time: 60,
      },
    });

    expect(rendered).toMatchSnapshot();
  });

  test("prompt, options", () => {
    const rendered = render({
      prompt: "mock-prompt",
      options: {
        use_cache: true,
        wait_for_model: false,
      },
    });

    expect(rendered).toMatchSnapshot();
  });

  test("all options", () => {
    const rendered = render({
      prompt: "mock-prompt",
      past_user_inputs: ["input1", "input2"],
      generated_responses: ["response1", "response2"],
      parameters: {
        min_length: 10,
        max_length: 100,
        top_k: 50,
        top_p: 0.9,
        temperature: 0.7,
        repetition_penalty: 1.2,
        max_time: 60,
      },
      options: {
        use_cache: true,
        wait_for_model: false,
      },
    });

    expect(rendered).toMatchSnapshot();
  });
});
