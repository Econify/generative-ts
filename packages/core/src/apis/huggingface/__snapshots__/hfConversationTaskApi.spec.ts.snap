// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HfConversationalTaskOptions: $prompt 1`] = `
{
  "inputs": "mock-prompt",
}
`;

exports[`HfConversationalTaskOptions: $prompt, generated_responses 1`] = `
{
  "generated_responses": [
    "response1",
    "response2",
  ],
  "inputs": "mock-prompt",
}
`;

exports[`HfConversationalTaskOptions: $prompt, options 1`] = `
{
  "inputs": "mock-prompt",
  "options": {
    "use_cache": true,
    "wait_for_model": false,
  },
}
`;

exports[`HfConversationalTaskOptions: $prompt, parameters 1`] = `
{
  "inputs": "mock-prompt",
  "parameters": {
    "max_length": 100,
    "max_time": 60,
    "min_length": 10,
    "repetition_penalty": 1.2,
    "temperature": 0.7,
    "top_k": 50,
    "top_p": 0.9,
  },
}
`;

exports[`HfConversationalTaskOptions: $prompt, past_user_inputs 1`] = `
{
  "inputs": "mock-prompt",
  "past_user_inputs": [
    "input1",
    "input2",
  ],
}
`;

exports[`HfConversationalTaskOptions: $prompt, past_user_inputs, generated_responses 1`] = `
{
  "generated_responses": [
    "response1",
    "response2",
  ],
  "inputs": "mock-prompt",
  "past_user_inputs": [
    "input1",
    "input2",
  ],
}
`;

exports[`HfConversationalTaskOptions: all options 1`] = `
{
  "generated_responses": [
    "response1",
    "response2",
  ],
  "inputs": "mock-prompt",
  "options": {
    "use_cache": true,
    "wait_for_model": false,
  },
  "parameters": {
    "max_length": 100,
    "max_time": 60,
    "min_length": 10,
    "repetition_penalty": 1.2,
    "temperature": 0.7,
    "top_k": 50,
    "top_p": 0.9,
  },
  "past_user_inputs": [
    "input1",
    "input2",
  ],
}
`;
